src/emulation/mod.rs

```diff
@@ -186,3 +191,16 @@ mod run_against_protocol {
         Ok(())
     }
 }
+
+pub fn run_against_protocols(
+    context: Context,
+    executable: &Path,
+    expected: Vec<Protocol>,
+) -> R<TestResults> {
+    Ok(TestResults(
+        expected
+            .into_iter()
+            .map(|expected| run_against_protocol(context.clone(), executable, expected))
+            .collect::<R<Vec<TestResult>>>()?,
```

Comment:
Is the explicit type instantiation required here?

Reply:
Yes. I suspect because of the `?`. The `?` implicitly converts errors (inside `Result::Err`) to different errors (using `From`). So -- without the turbofish -- the result of `collect` would be allowed to be any type `Result<Vec<TestResult>, E>` where `E` can be converted to our error type (`Box<std::error::Error>`, see [here](https://github.com/Originate/check-protocols/blob/master/src/lib.rs#L15)). The turbofish pins down the error type.

It could be shortened to `::<R<_>>`. Do you think that'd be better?

Reply:
I see, that makes sense. I would keep the full type in that case!